cmd/kubeadm/app/preflight/checks.go

```diff
@@ -864,6 +904,7 @@ func RunInitMasterChecks(execer utilsexec.Interface, cfg *kubeadmapi.MasterConfi
 		HTTPProxyCheck{Proto: "https", Host: cfg.API.AdvertiseAddress},
 		HTTPProxyCIDRCheck{Proto: "https", CIDR: cfg.Networking.ServiceSubnet},
 		HTTPProxyCIDRCheck{Proto: "https", CIDR: cfg.Networking.PodSubnet},
+		ImagePullCheck{Images: criInterfacer, ImageList: images.GetAllImages(cfg)},
```

Comment:
Idea, can we create a dedicated `RunPullImagesCheck` function that we in kubeadm init run _after all the init checks have succeeded_? That way we won't even start pulling if something else is really wrong, but reuse the preflight framework anyway.

Reply:
^ I like this
Should we allow the ImagePullCheck to be skippable if it is now part of the primary logic?
Since `crictl` is new, there may be users who do not have it and are not using docker.

Reply:
> Should we allow the ImagePullCheck to be skippable if it is now part of the primary logic?

Yes, that's one of the reasons we're hosting it as a preflight check, you can set the ignore preflight errors flag and boom you won't be bothered with this anymore if you don't like it

Reply:
Sure, this is better than shoving it somewhere in the middle. 

We'll also want to think about a list of images needed for joining.